data/migrations/20190110190931_recipe.js

```diff
@@ -0,0 +1,12 @@
+
+exports.up = function (knex, Promise) {
+    return knex.schema.createTable('recipe', table => {
+        table.increments('id');
+        table.string('name').notNullable();
+        table.foreign('dish_id').references('id').on('dish');
```

Comment:
Does it work to use `foreign` on a key that doesn't exist yet? I'm not sure to be honest, but here's this for more reading: https://knexjs.org/#Schema-foreign